scipy/stats/_stats_py.py

```diff
@@ -221,10 +221,21 @@ def _broadcast_shapes_with_dropped_axis(a, b, axis):
         lambda x: x, n_samples=1, n_outputs=1, too_small=0, paired=True,
         result_unpacker=lambda x: (x,), kwd_samples=['weights'])
 def gmean(a, axis=0, dtype=None, weights=None):
-    """Compute the geometric mean along the specified axis.
+    r"""Compute the weighted geometric mean along the specified axis.
 
-    Return the geometric average of the array elements.
-    That is:  n-th root of (x1 * x2 * ... * xn)
+    The weighted geometric mean of the array :math:`a_i` associated to weights
+    :math:`w_i` is:
+
+    .. math::
+
+        \exp \left( \frac{ \sum_{i=1}^n w_i \log a_i }{ \sum_{i=1}^n w_i }
+                   \right) \, ,
+
+    and, with equal weights, it falls backs to:
```

Comment:
I think fall back is not really suitable.
```suggestion
    and, with equal weights, it gives:
```